packages/app-desktop/gui/NoteListControls/NoteListControls.tsx

```diff
@@ -48,11 +69,68 @@ export default function NoteListControls(props: Props) {
 		void CommandService.instance().execute('newNote');
 	}
 
+	function onSortOrderFieldButtonClick() {
+		void CommandService.instance().execute(NOTES_SORT_ORDER_SWITCH);
+	}
+
+	function onSortOrderReverseButtonClick() {
+		void CommandService.instance().execute(NOTES_SORT_ORDER_TOGGLE_REVERSE);
+	}
+
+	function sortOrderFieldTooltip() {
+		const term1 = CommandService.instance().label(NOTES_SORT_ORDER_SWITCH);
+		const field = Setting.value(SETTING_FIELD);
+		const term2 = Note.fieldToLabel(field);
+		const term3 = Note.fieldToLabel(notesSortOrderNextField(field));
+		return `${term1}:\n ${term2} -> ${term3}`;
+	}
+
+	function sortOrderFieldIcon() {
+		const field = Setting.value(SETTING_FIELD);
+		const iconMap: any = {
+			user_updated_time: 'far fa-calendar-alt',
+			user_created_time: 'far fa-calendar-plus',
+			title: 'fas fa-font',
+			order: 'fas fa-wrench',
+		};
+		return `${iconMap[field] || iconMap['title']} ${field}`;
+	}
+
+	function sortOrderReverseIcon() {
+		return Setting.value(SETTING_REVERSE) ? 'fas fa-long-arrow-alt-up' : 'fas fa-long-arrow-alt-down';
+	}
+
+	function sortOrderButtonsVisible() {
+		let visible = Setting.value('notes.sortOrder.buttonsVisible');
```

Comment:
Also don't call Setting.value() from components, because if the value is changed the component should be re-rendered, but it won't be when using Setting.value. Instead pass the values you need to the props using Redux connect, and the state - state.settings['notes.sortOrder.buttonsVisible']

Reply:
As a come-lately committer, I felt reluctant to touch React component's state and props when not needed. If you don't mind, it would be really better.

By the way, since NoteListControl changes when any 'sort.noteOrderXXX' setting changes, the code in this commit works properly as expected.

Reply:
finished. e9dcc3b